### PR TITLE
make: delete a compiling flag '-Wno-sign-compare'

### PR DESCRIPTION
**Pull request by wudping**
delete the '-Wno-sign-compare' of both CFLAGS and CXXFLAGS in file configure.ac, and work out the compiling error in comparison between signed and unsigned integer expression.

### CODE REVIEW

**Comment by uartie** [comment]
Can you fix the following things in the commit message:
1. Separate subject from body with a blank line
2. Wrap the body to about 72 characters or so
3. Specify in the body: "Fixes #350"

**Comment by uartie** [comment]
I still get a compile error here:
```
In file included from vaapidecpicture.h:25:0,
from vaapidecpicture.cpp:26:
../vaapi/vaapipicture.h: In instantiation of 'bool YamiMediaCodec::VaapiPicture::render(std::vector<O>&) [with O = std::pair<std::tr1::shared_ptr<YamiMediaCodec::VaapiBufObject>, std::tr1::shared_ptr<YamiMediaCodec::VaapiBufObject> >]':
vaapidecpicture.cpp:53:44:   required from here
../vaapi/vaapipicture.h:112:23: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
for (int i = 0; i < objects.size(); i++)
^
```

**Comment by uartie** [comment]
Be sure to enable all the codecs, v4l2, and tests in configure to ensure all the sign-compare errors are gone.  Because I'm still getting the following in addition to above errors:

```
vaapiencoder_vp8.cpp: In member function 'bool YamiMediaCodec::VaapiEncoderVP8::fill(VAQMatrixBufferVP8*) const':
vaapiencoder_vp8.cpp:213:19: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
for (i = 0; i < N_ELEMENTS(qMatrix->quantization_index); i++) {
^
vaapiencoder_vp8.cpp:217:19: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
for (i = 0; i < N_ELEMENTS(qMatrix->quantization_index_delta); i++) {
^
```

...and here

```
vaapiencoder_hevc.cpp: In function 'uint32_t YamiMediaCodec::log2(uint32_t)':
vaapiencoder_hevc.cpp:131:16: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
while (num > (1 << ret))
^
vaapiencoder_hevc.cpp: In member function 'void YamiMediaCodec::VaapiEncStreamHeaderHEVC::generateCodecConfig()':
vaapiencoder_hevc.cpp:352:27: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
for (int i = 0; i < headers.size(); i++) {
^
vaapiencoder_hevc.cpp: In member function 'Encode_Status YamiMediaCodec::VaapiEncPictureHEVC::getOutput(VideoEncOutputBuffer*, std::vector<std::tr1::function<int()> >&)':
vaapiencoder_hevc.cpp:756:27: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
for (int i = 0; i < functions.size(); i++) {
^
```
...and here

```
In file included from ../tests/vppinputoutput.h:28:0,
                 from ../tests/vppinputdecode.h:27,
                 from vppinputdecode.cpp:22:
../tests/videopool.h: In instantiation of ‘YamiMediaCodec::VideoPool<T>::VideoPool(std::deque<std::tr1::shared_ptr<_Tp> >&) [with T = VideoFrame]’:
../tests/videopool.h:34:74:   required from ‘static std::tr1::shared_ptr<YamiMediaCodec::VideoPool<T> > YamiMediaCodec::VideoPool<T>::create(std::deque<std::tr1::shared_ptr<_Tp> >&) [with T = VideoFrame]’
../tests/vppinputoutput.h:120:41:   required from here
../tests/videopool.h:53:31: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
             for (int i = 0; i < m_holder.size(); i++) {
```

**Comment by wudping** [comment]
Hi Uartie, thank you very much for your review. I don't get the compiling errors because I don't enable those functions. It's my fault.

File: decoder/vaapidecoder_fake.cpp [hunk 1]
```diff
@@ -45,7 +45,7 @@ Decode_Status VaapiDecoderFake::start(VideoConfigBuffer * buffer)
 {
     DEBUG("VP9: start() buffer size: %d x %d", buffer->width,
           buffer->height);
-    Decode_Status status;
```
Comment (xuguangxin) [inline decoder/vaapidecoder_fake.cpp]: you can remove this if no one use it

**Comment by xuguangxin** [comment]
ok to me, we will rebase and test it after 0.3.1 released

**Comment by xuguangxin** [comment]
Hi Dongping:
Could you rebase this and do regression test for all codecs?

**Comment by xuguangxin** [comment]
close this since dongpin send #424 for new pull request

**Pull request closed without merging**